File: oa-portal/problems/c24-counting-divisors/starters/main.java
import java.io.*;
import java.util.*;

public class Main {

    // All queries share one small universe (x <= 1e6) - precompute once.
    static int[] divisorCounts(int[] xs) {
        // write your code here

        return new int[xs.length];
    }

    public static void main(String[] args) throws IOException {
        FastReader in = new FastReader();
        int n = in.nextInt();
        int[] xs = new int[n];
        for (int i = 0; i < n; i++) xs[i] = in.nextInt();

        StringBuilder sb = new StringBuilder();
        for (int v : divisorCounts(xs)) sb.append(v).append('\n');
        System.out.print(sb);
    }

    static class FastReader {
        private final DataInputStream in = new DataInputStream(new BufferedInputStream(System.in, 1 << 16));
        int nextInt() throws IOException { return (int) nextLong(); }
        long nextLong() throws IOException {
            int b = in.read();
            while (b != '-' && (b < '0' || b > '9')) b = in.read();
            boolean neg = b == '-';
            if (neg) b = in.read();
            long v = 0;
            while (b >= '0' && b <= '9') { v = v * 10 + (b - '0'); b = in.read(); }
            return neg ? -v : v;
        }
        String next() throws IOException {
            int b = in.read();
            while (b <= ' ') b = in.read();
            StringBuilder sb = new StringBuilder();
            while (b > ' ') { sb.append((char) b); b = in.read(); }
            return sb.toString();
        }
    }
}

File: oa-portal/problems/c24-counting-divisors/starters/main.cpp
#include <bits/stdc++.h>
using namespace std;

// All queries share one small universe (x <= 1e6) - precompute once.
vector<int> divisorCounts(const vector<int>& xs) {
    // write your code here

    return vector<int>(xs.size(), 0);
}

int main() {
    int n;
    if (scanf("%d", &n) != 1) return 0;
    vector<int> xs(n);
    for (auto &v : xs) scanf("%d", &v);

    string out;
    for (int v : divisorCounts(xs)) { out += to_string(v); out += '\n'; }
    fwrite(out.data(), 1, out.size(), stdout);
    return 0;
}
